src/main/java/com/redhat/cloud/policies/app/rest/utils/PagingUtils.java

```diff
@@ -93,23 +93,22 @@ public static Pager extractPager(UriInfo uriInfo) {
                 String column = columns.get(i);
                 Sort.Direction direction = Sort.Direction.Ascending;
                 if (directions != null && i < directions.size()) {
-                    switch(directions.get(i).toLowerCase()) {
-                        case "asc":
-                            direction = Sort.Direction.Ascending;
-                            break;
-                        case "desc":
-                            direction = Sort.Direction.Descending;
-                            break;
-                        default:
-                            throw new IllegalArgumentException("Unexpected sort order found: [" + columns.get(i) + "]");
-                    }
+                    String dir = directions.get(i);
+                    direction = getDirection(dir, columns.get(i));
                 }
                 pageBuilder.addSort(column, direction);
             }
         }
         else {
             // default sort is by mtime descending, so that newest end up on top
-            pageBuilder.addSort("mtime",Sort.Direction.Descending);
+            Sort.Direction direction = Sort.Direction.Descending;
+
+            // check if the user specified a sort order (but no column, so mtime is meant)
+            if (directions!=null && directions.size()>0) {
+                direction = getDirection(directions.get(0), "mtime");
+            }
+
+            pageBuilder.addSort("mtime",direction);
```

Comment:
The inconvenient with setting the sort order in this function is that we don't have `mtime` in the trigger history.

So (like the original issue), the contents of the following calls are the same:
- `curl 'https://ci.foo.redhat.com:1337/api/policies/v1.0/policies/c8d5cce5-7e04-4386-9875-8ccc0b27687b/history/trigger?limit=50&offset=0&sortOrder=desc'` 
- `curl 'https://ci.foo.redhat.com:1337/api/policies/v1.0/policies/c8d5cce5-7e04-4386-9875-8ccc0b27687b/history/trigger?limit=50&offset=0&sortOrder=asc'`
- `curl 'https://ci.foo.redhat.com:1337/api/policies/v1.0/policies/c8d5cce5-7e04-4386-9875-8ccc0b27687b/history/trigger?limit=50&offset=0'`

Perhaps we should pass some defaults to this function?

Reply:
I don't think that is an issue with the term 'mtime'. In https://github.com/RedHatInsights/policies-ui-backend/blob/9e9ad0d7fdbd7634045c591e051a4a42852c6cc5/src/main/java/com/redhat/cloud/policies/app/rest/PolicyCrudService.java#L978 it uses 'ctime' if the column name is not either 'id' or 'name'.

I think I know what you see: you use `sortOrder`- that is engine speak. The ui-backend uses `sortDirection` , which some un-named person introduced for getPolicies some time ago :-)

Reply:
My bad!